src/Common/src/Interop/OSX/Interop.CoreFoundation.CFData.cs

```diff
@@ -61,10 +68,13 @@ internal static unsafe bool TryCFWriteData(SafeCFDataHandle cfData, Span<byte> d
                     return false;
                 }
 
-                byte* dataBytes = CFDataGetBytePtr(cfData);
-                fixed (byte* destinationPtr = &MemoryMarshal.GetReference(destination))
+                if (length > 0)
```

Comment:
Minor, but how about moving the above if block into this one?  If length <= 0, then destination.Length will never be < length.